src/ReadoutApplication.cpp

```diff
@@ -429,9 +432,12 @@ ReadoutApplication::generate_modules(const confmodel::Session* session) const
   frag_aggr.set_objs("inputs", { &fa_net_obj, &frag_queue_obj });
   frag_aggr.set_objs("outputs", fa_output_objs);
 
-  modules.push_back(obj_fac.get_dal<confmodel::DaqModule>(frag_aggr.UID()));
+  auto dal_obj = obj_fac.get_dal<confmodel::DaqModule>(frag_aggr.UID());
+  module_objects.push_back(&dal_obj->config_object());
 
-  return modules;
+  auto app_obj = obj_fac.get_dal<ReadoutApplication>(UID())->config_object();
+  app_obj.set_objs("modules", module_objects);
+  configuration().update<ReadoutApplication>({UID()}, {}, {});
```

Comment:
It doesn't feel great to expose this call to "usercode". 
How about adding a helper method to `ConfigObjectFactory` to call `update<ReadoutApplication>({UID()}, {}, {});`?
(yes, that requires tweaking the object factory a little, but it's probably doable without much work...)